lib/marshal.js

```diff
@@ -83,7 +83,11 @@ Marshal = (function () {
     }
     else if (small === -1) {
       var large = this.buffer.readInt8(this._index++);
-      return large;
+      if (large >= 0 && large <= 127) {
+        return large - 256;
```

Comment:
@clayzermk1 even though subtracting 256 from an Int8 in static language would not be correct, I think this is a perfectly valid thing to do in JavaScript.